### PR TITLE
[feature] external table resource

### PR DESCRIPTION
**Pull request by fevin86**
This introduce resource creation and deletion for external table.

## Test Plan
* [x] acceptance tests

## References
https://docs.snowflake.com/en/sql-reference/sql/create-external-table.html

### CODE REVIEW

**Review by ryanking — CHANGES_REQUESTED** [review]

**Comment by fevin86** [comment]
@ryanking can you please have another look at this PR?

**Comment by yonidavidson** [comment]
Hi, does it expose notification_channel so I can use it to register to s3 events?

**Comment by ryanking** [comment]
@fevin86 can you merge the main branch and resolve conflicts?

**Review by ryanking — CHANGES_REQUESTED** [review]
@fevin86 can you merge the main branch and resolve conflicts?

**Comment by fevin86** [comment]
@ryanking, I have merged the main and fixed the build. Can you please have a look again?

File: pkg/resources/external_table.go [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+					ForceNew:    true,
+				},
+			},
+		},
+	},
+	"location": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies a comment for the external table.",
```
Comment (ryanking) [inline pkg/resources/external_table.go:71]: You mean location?

File: pkg/resources/external_table.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+		Type:        schema.TypeList,
+		Optional:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		ForceNew:    true,
+		Description: "Specifies any partition columns to evaluate for the external table.",
+	},
+	"refresh_on_create": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies to retain the access permissions from the original table when an external table is recreated using the CREATE OR REPLACE TABLE variant",
```
Comment (ryanking) [inline pkg/resources/external_table.go:96]: I think this was copied from _auto_refresh below.

File: pkg/resources/external_table.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+		return nil, fmt.Errorf("3 fields allowed")
+	}
+
+	externalTableResult := &externalTableID{
+		DatabaseName:      lines[0][0],
+		SchemaName:        lines[0][1],
+		ExternalTableName: lines[0][2],
+	}
+	return externalTableResult, nil
+}
```
Comment (ryanking) [inline pkg/resources/external_table.go:186]: I know that there are several copies of this code already, but at some point we should probably stop copying it and instead create something reusable.

File: pkg/resources/external_table.go [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+		Description: "Name of the role that owns the external table.",
+	},
+}
+
+func ExternalTable() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateExternalTable,
+		Read:   ReadExternalTable,
+		Delete: DeleteExternalTable,
+		Exists: ExternalTableExists,
```
Comment (ryanking) [inline pkg/resources/external_table.go:133]: You should be able to drop this.

**Review by ryanking — CHANGES_REQUESTED** [review]

File: pkg/resources/external_table.go (hunk 1 above)
Comment (fevin86) [reply pkg/resources/external_table.go]: fixed

File: pkg/resources/external_table.go (hunk 2 above)
Comment (fevin86) [reply pkg/resources/external_table.go]: fixed

File: pkg/resources/external_table.go (hunk 4 above)
Comment (fevin86) [reply pkg/resources/external_table.go]: removed

**Review by ryanking — APPROVED** [review]

**Comment by ryanking** [comment]
@fevin86 see test failures

**Comment by fevin86** [comment]
@ryanking can you kick off the integration test again?

**Pull request merged**